core/outbox.go

```diff
@@ -120,9 +121,11 @@ func (ob *Outbox) Send(ctx context.Context, from, to address.Address, value type
 		return cid.Undef, errors.Wrap(err, "failed to add message to outbound queue")
 	}
 
-	err = ob.publisher.Publish(ctx, signed, height)
-	if err != nil {
-		return cid.Undef, err
+	if publish {
```

Comment:
Blocking: this doesn't do what you want. This will prevent the message reaching the message pool too: see `defaultMessagePublisher`. This parameter needs to be passed to the publisher. Given that, `broadcast` might be a better name for the parameter, though it's all just a matter of scopes.

Unfortunately this is a little difficult to integration test without adding an interface to abstract over the `pubsub.Publisher` so you can mock it. But you can improve the `mockPublisher` to record what value was the flag that it received.

Alternatively, the outbox could be refactored to hold a reference to (an interface to) the message pool, essentially inlining `defaultMessagePublisher` in the outbox. I'm ok with that too so long as the concrete message pool type isn't referenced. Slight preference for the first way.

Reply:
I did sort of both: I made  node’s defaultMessagePublisher’s pubsub.Publisher refer instead to an interface/ It calls just one function on the publisher, so that the testing is more complete. We are just testing basic logic, so if you think it's too much I'll revert it.